util/errors/errors.go

```diff
@@ -37,7 +37,8 @@ func IsTransientErr(err error) bool {
 		apierr.IsServiceUnavailable(err) ||
 		isTransientEtcdErr(err) ||
 		matchTransientErrPattern(err) ||
-		errors.Is(err, NewErrTransient(""))
+		errors.Is(err, NewErrTransient("")) ||
+		strings.Contains(err.Error(), "upper: no more rows in ")
```

Comment:
yeah, the root cause is as mentioned above. and I will have a test for the function `isTransientSqbErr `.

Reply:
Where did this error occur? Is there an example to reproduce?

Reply:
yeah, I run it local with `make start UI=true PROFILE=mysql ALWAYS_OFFLOAD_NODE_STATUS=true`, and submit a workflow and stop it when one step succeed, after it is archived, then retry it. (remove this error caght, it will retry failed at here https://github.com/argoproj/argo-workflows/blob/main/workflow/controller/controller.go#L854)

Reply:
I haven't reproduced it, using the latest version. What version are you using?

Reply:
It may occur in old version：https://github.com/argoproj/argo-workflows/issues/2333

Reply:
Yeah，I have checked this issue, and I got the error with evrsion v3.5.0, and the latest commit it also has this error. I test it at locally according to  https://argo-workflows.readthedocs.io/en/latest/running-locally/, and do you add the env `ALWAYS_OFFLOAD_NODE_STATUS=true`? @shuangkun

Reply:
I use `make start UI=true PROFILE=mysql ALWAYS_OFFLOAD_NODE_STATUS=true` run worflow-controller and server and I use latest.

Reply:
And I just pull my branch `heidongxianhua:fix_retry_large_archived_wf` and  rebased by the branch `[argoproj:main](https://github.com/argoproj/argo-workflows/tree/main)`. If  i add the error check `https://github.com/argoproj/argo-workflows/pull/12741/files#diff-c2fc14b8209201e80dbf52f4293639d125ce3b6df398ebc93f5ee4e1abf05842R41`, it will run ok and the server.log contains this, but there is no error in controller.log:
![image](https://github.com/argoproj/argo-workflows/assets/22924371/36093ced-9108-49cf-87f9-41699602fda1)
 
But if i delete this check (run `make clean` and `make start UI=true PROFILE=mysql ALWAYS_OFFLOAD_NODE_STATUS=true`), we will get the error in controller.log, and it lost the node:
![image](https://github.com/argoproj/argo-workflows/assets/22924371/ecaa0c3f-b5a3-4355-90cc-e72fc3abc338)
![image](https://github.com/argoproj/argo-workflows/assets/22924371/6620919b-2b19-4a80-9b01-5b10a6c0ac57)
@shuangkun

Reply:
OK, thanks! I reproduced it.  LGTM